### PR TITLE
Use built-in XML editor support for VSWin

### LINKED ISSUE #20: Right-clicking on the project in VS solution explorer and selecting 'Edit Project File' doesn't let the .csproj file to open
After the Build (in both Debug/Release mode, where `MonoDevelop.MSBuild.Editor.VisualStudio` is _Set as StartUp Project_) and running the instance of the Visual Studio with this extension installed, if done the following:

On creating a simple Console App (.NET Core) project:
image

Selecting Edit Project File:
image

This dialogue box appears:
image

Clicking yes/no doesn't matter. It doesn't open the .csproj file

Only way to do so is first close the project and then go to _file->open->file->'select'_ csproj file

Closing the project before opening its csproj file is important. If **not** closed before, and then try to open the csproj file through _file->open->file_, then the same yes/no dialogue box reappears.

Visual Studio version used: _16.2.2_

### PR DESCRIPTION
**Pull request by wjk**
This PR avoids the problem described in #42, as the MSBuild support now builds on top of the built-in XML editor, instead of attempting to replace it. With this PR applied, right-click > Edit Project File now opens the editor as expected, while enabling rich MSBuild IntelliSense, Quick Info, and syntax coloring.

Fixes #20.

### CODE REVIEW

**Comment by KirillOsenkov** [comment]
We need to be cognizant of the fact that we're also planning to ship this in VSMac at some point, which will use MonoDevelop.Xml as its primary XML language service. So we need to strike a delicate balance where we expect MonoDevelop.Xml to be present on Mac, but not on Windows.

**Comment by KirillOsenkov** [comment]
Now that I've read the PR I see that the changes are confined to the VSWin-specific files, and seem to make sense to me. Let's see what Mikayla says, maybe the intention was to completely replace the VS XML editor experience at least for MSBuild files, and not sit on top of it. 

Also we need to audit the Xml.Editor assembly to see if it might export MEF things that are needed for MSBuild to work fine. If it only exports things for XML support, that's fine. But if not, we'd need to split the Xml.Editor assembly or move some types around. Basically we need to remember that it will ship in two host IDEs, one having the XML and the other not.

Of course long term I'd like to replace the existing XML language service in Visual Studio as it really needs modernization. But this is a whole can of worms I don't want to open right now.

**Comment by mhutch** [comment]
This will for sure break:

* structure navigation
* navigating between highlighted references
* tag and attribute commit behaviors

Yes, we could split stuff up into more assemblies but I have concerns about mixing two different language services in this way due to the perf impact and the risk of bugs from interactions between the language services' commands, and this will only get worse once we add stuff like editorconfig support and autoformatting.

**Comment by wjk** [comment]
@mhutch The main issue I found while implementing this is the MonoDevelop.Xml projects are in a submodule, and so I had to find a way to implement what was needed without altering anything in it, which would greatly complicate testing of this PR. However, that being said, if we wanted we could move the code for those two projects into this repo, and then I can work on modifying MonoDevelop.Xml.Editor to work properly in VS. Let me know if you want me to take this approach.

Also keep in mind @lohitakshgupta's comment in https://github.com/mhutch/MonoDevelop.MSBuildEditor/issues/20#issuecomment-522802631. Although it's a bit hard to parse due to a number of probable typos, I have found through experimentation that I _must_ inherit from the built-in XML editor service, or Visual Studio will refuse to edit a project file without unloading it first. The code responsible for loading the editor upon right-click/Edit Project hard-codes the built-in XML editor GUID, and I have found no way to override that requirement (not even by reimplementing the command from scratch based on decompiling the originals).

**Comment by wjk** [comment]
Also, @KirillOsenkov, I'm just curious: What about the built-in XML editor do you want to improve or replace?

**Comment by KirillOsenkov** [comment]
@wjk the VS XML editor is a very old codebase that no longer has a clear owner and was written way before the new editor came into existence. It's probably the oldest managed component in Visual Studio. For maintainability and evolving it into the future I would want to replace it with a more modern alternative, such as MonoDevelop.Xml.

**Comment by wjk** [comment]
@mhutch @KirillOsenkov I'd appreciate some eyes on another implementation of this feature in #48. Thanks!

**Comment by mhutch** [comment]
Thanks for the contribution, it was definitely good to consider this approach! However, I'm closing this as I'm porting the extension to VS Code and this will not be a viable route there.
